tests/providers/amazon/aws/hooks/test_hooks_signature.py

```diff
@@ -187,10 +187,10 @@ def method2(self, spam: str):
     if not hooks:
         pytest.skip(reason=f"Module {hook_module!r} doesn't contain subclasses of `AwsGenericHook`.")
 
-    errors = []
+    errors: list[str] = []
     for hook, hook_name in hooks:
         is_valid, msg = validate_hook(hook, hook_name, hook_module)
-        if not is_valid:
+        if not is_valid and msg:
             errors.append(msg)
```

Comment:
```suggestion
    validation_results = (validate_hook(hook, hook_name, hook_module) for hook, hook_name in hooks)
    errors = [
        message
        for valid, message in validation_results
        if not valid and message
    ]
```

Reply:
Why not :). We can even inline valudation_results it seems.